js_modules/dagit/packages/core/src/runs/StepComputeLogsDialog.tsx

```diff
@@ -0,0 +1,143 @@
+import {
+  ILogCaptureInfo,
+  IRunMetadataDict,
+  RunMetadataProvider,
+  matchingComputeLogKeyFromStepKey,
+} from './RunMetadataProvider';
+import {titleForRun} from './RunUtils';
+
+export const StepComputeLogsDialog: React.FC<{
+  runId: string;
+  stepKeys: string[];
+  isOpen: boolean;
+  onClose: () => void;
+}> = ({runId, stepKeys, isOpen, onClose}) => {
+  return (
+    <Dialog
+      isOpen={isOpen}
+      style={{width: '80vw'}}
+      canOutsideClickClose
+      canEscapeKeyClose
+      onClose={onClose}
+    >
+      <DialogBody>
+        {isOpen ? (
+          <LogsProvider key={runId} runId={runId}>
+            {(logs) => (
+              <RunMetadataProvider logs={logs}>
+                {(metadata) => (
+                  <StepComputeLogsModalContent
+                    runId={runId}
+                    metadata={metadata}
+                    stepKeys={stepKeys}
+                  />
+                )}
+              </RunMetadataProvider>
+            )}
+          </LogsProvider>
+        ) : (
+          ''
+        )}
+      </DialogBody>
+      <DialogFooter topBorder>
+        <Button intent="primary" onClick={onClose}>
+          Done
+        </Button>
+      </DialogFooter>
+    </Dialog>
+  );
+};
+
+export const StepComputeLogsModalContent: React.FC<{
+  runId: string;
+  stepKeys: string[];
+  metadata: IRunMetadataDict;
+}> = ({runId, stepKeys, metadata}) => {
+  const supportsCapturedLogs = useSupportsCapturedLogs();
+  const [logType, setComputeLogType] = useState<LogType>(LogType.stdout);
+  const [computeLogUrl, setComputeLogUrl] = React.useState<string | null>(null);
+  const [computeLogFileKey, setComputeLogFileKey] = React.useState<string>('');
+
+  const logCaptureInfo: ILogCaptureInfo | undefined =
+    metadata.logCaptureSteps && computeLogFileKey in metadata.logCaptureSteps
+      ? metadata.logCaptureSteps[computeLogFileKey]
+      : undefined;
+
+  React.useEffect(() => {
+    if (!stepKeys?.length || computeLogFileKey) {
+      return;
+    }
+
+    if (metadata.logCaptureSteps) {
+      const logFileKeys = Object.keys(metadata.logCaptureSteps);
+      const selectedLogKey = logFileKeys.find((logFileKey) => {
+        return stepKeys.every(
+          (stepKey) =>
+            metadata.logCaptureSteps &&
+            metadata.logCaptureSteps[logFileKey].stepKeys.includes(stepKey),
+        );
+      });
+      setComputeLogFileKey(selectedLogKey || logFileKeys[0]);
+    } else if (!stepKeys.includes(computeLogFileKey)) {
+      const matching = matchingComputeLogKeyFromStepKey(metadata.logCaptureSteps, stepKeys[0]);
+      matching && setComputeLogFileKey(matching);
+    } else if (stepKeys.length === 1 && computeLogFileKey !== stepKeys[0]) {
+      const matching = matchingComputeLogKeyFromStepKey(metadata.logCaptureSteps, stepKeys[0]);
+      matching && setComputeLogFileKey(matching);
+    }
+  }, [stepKeys, computeLogFileKey, metadata.logCaptureSteps, setComputeLogFileKey]);
```

Comment:
I'm always wary of calling `setState` inside of `useEffect` but I trust its probably necessary here